### PR TITLE
Node 18 Upgrade

### PR DESCRIPTION
**Pull request by SyedWasiHaider**
Upgrading to node 18.

https://docs.google.com/document/d/1x7NYmsTIhZAJMxL9VTcbWEZanPcK_RoRe_rVSjrqcUE/edit

## Testing

_Include any additional information about the testing you have completed to
ensure your changes behave as expected. For a speedy review, please check
any of the tasks you completed below during your testing._

- [ ] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/browser-destinations/package.json [hunk 1]
```diff
@@ -19,8 +19,8 @@
     "build": "yarn clean && yarn build-ts && yarn build-cjs && yarn build-web",
     "build-ts": "yarn tsc -b tsconfig.build.json",
     "build-cjs": "yarn tsc -p ./tsconfig.build.json -m commonjs --outDir ./dist/cjs/",
-    "build-web": "NODE_ENV=production ASSET_ENV=production yarn webpack -c webpack.config.js",
-    "build-web-stage": "NODE_ENV=production ASSET_ENV=stage yarn webpack -c webpack.config.js",
+    "build-web": "NODE_ENV=production ASSET_ENV=production NODE_OPTIONS=--openssl-legacy-provider yarn webpack -c webpack.config.js",
+    "build-web-stage": "NODE_ENV=production ASSET_ENV=stage NODE_OPTIONS=--openssl-legacy-provider yarn webpack -c webpack.config.js",
```
Comment (NolanChan) [inline packages/browser-destinations/package.json:23]: just curious about what `--openssl-legacy-provider` does for action-destinations?

**Review by dlasky — APPROVED** [review]
lgtm, would like 2nd 👁️ from actions

**Review by NolanChan — APPROVED** [review]
Thankful nothing crazy was involved :D

File: packages/browser-destinations/package.json (hunk 1 above)
Comment (dlasky) [reply packages/browser-destinations/package.json:23]: some webpack bug

https://github.com/webpack/webpack/issues/14532

**Review by prayansh — APPROVED** [review]
🚢

**Pull request merged**